### DISCUSSION TITLE
strongly typed signalR with streaming

### DISCUSSION THREAD

**Discussion by gesho** [dotnet/aspnetcore#54276] [Q&A]
is it possible to have streaming with strongly typed hub using `IAsyncEnumerable`?

for example when I try this sort of IHub, project
- builds
- runs
-  crashes on first connection with message

may be I should be wrapping return in `Task<>`?

```cs
    public interface IHub
    {
        Task PostToClientAsync(Message message);

        async IAsyncEnumerable<Message> StreamToClientAsync(IEnumerable<Message> messages)
        {
            foreach (var message in messages)
            {
                await Task.Delay(1000);
                yield return message;
            }
        }

    }
```

> Error: Cannot generate proxy implementation for '***.Hub.IHub.StreamToClientAsync'. All client proxy methods must return 'System.Threading.Tasks.Task' or 'System.Threading.Tasks.Task<T>'.
fail: Microsoft.AspNetCore.SignalR.Internal.DefaultHubDispatcher[8]
      Failed to invoke hub method 'PostToHubAsync'.
      System.InvalidOperationException: Cannot generate proxy implementation for ***.Hub.IHub.StreamToClientAsync'. All client proxy methods must return 'System.Threading.Tasks.Task' or 'System.Threading.Tasks.Task<T>'.

**Comment by gfoidl** [comment]
@BrennanConroy please have a look here.

**Comment by BrennanConroy** [comment]
Looks like a bug. We probably didn't think about strongly-typed hubs when adding streaming support.
